src/contracts/IbexaTestKernel.php

```diff
@@ -0,0 +1,289 @@
+
+use Doctrine\Bundle\DoctrineBundle\DoctrineBundle;
+use Doctrine\DBAL\Connection;
+use FOS\JsRoutingBundle\FOSJsRoutingBundle;
+use Ibexa\Bundle\Core\IbexaCoreBundle;
+use Ibexa\Bundle\LegacySearchEngine\IbexaLegacySearchEngineBundle;
+use Ibexa\Contracts\Core\Persistence\TransactionHandler;
+use Ibexa\Contracts\Core\Repository;
+use Ibexa\Contracts\Core\Test\IbexaTestKernelInterface;
+use Ibexa\Contracts\Core\Test\Persistence\Fixture\YamlFixture;
+use Ibexa\Tests\Integration\Core\IO\FlysystemTestAdapter;
+use Ibexa\Tests\Integration\Core\IO\FlysystemTestAdapterInterface;
+use JMS\TranslationBundle\JMSTranslationBundle;
+use League\Flysystem\InMemory\InMemoryFilesystemAdapter;
+use Liip\ImagineBundle\LiipImagineBundle;
+use LogicException;
+use Psr\Log\NullLogger;
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\SecurityBundle\SecurityBundle;
+use Symfony\Bundle\TwigBundle\TwigBundle;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\HttpKernel\Kernel;
+
+/**
+ * @experimental
+ *
+ * Baseline test kernel that dependent packages can extend for their integration tests.
+ *
+ * ## Configuring the kernel
+ *
+ * It automatically exposes all Repository-based services for consumption in tests (marking them as public prevents
+ * them from being removed from test container). A minimal Symfony framework configuration is provided,
+ * along with Doctrine connection.
+ *
+ * To supply a different configuration, extend IbexaTestKernel::loadConfiguration() method.
+ *
+ * You can supply your own services (which is something you probably want) by extending IbexaTestKernel::loadServices().
+ *
+ * If you need even more control over how the container is built you can do that by extending the
+ * IbexaTestKernel::registerContainerConfiguration().
+ *
+ * ## Adding bundles
+ *
+ * Bundles can be added by extending IbexaTestKernel::registerBundles() method (just like in any Kernel).
+ *
+ * ## Exposing your services
+ *
+ * To add services to the test Kernel and make them available in tests via IbexaKernelTestCase::getServiceByClassName(),
+ * you'll need to extend IbexaTestKernel::getExposedServicesByClass() and / or IbexaTestKernel::getExposedServicesById()
+ * method.
+ *
+ * IbexaTestKernel::getExposedServicesByClass() is a simpler variant provided for services that are registered in
+ * service container using their FQCN.
+ *
+ * IbexaTestKernel::getExposedServicesById() is useful if your service is not registered as it's FQCN (for example,
+ * if you have multiple services for the same class / interface).
+ *
+ * If don't need the repository services (or not all), you can replace the IbexaTestKernel::EXPOSED_SERVICES_BY_CLASS and
+ * IbexaTestKernel::EXPOSED_SERVICES_BY_ID consts in extending class, without changing the methods above.
```

Comment:
Readability (line too long) and grammar (missing pronoun):
```suggestion
 * If you don't need the repository services (or not all of them), you can replace
 * the IbexaTestKernel::EXPOSED_SERVICES_BY_CLASS and IbexaTestKernel::EXPOSED_SERVICES_BY_ID constants
 * in the extending class, without changing the methods above.
```